WcaOnRails/app/models/competition.rb

```diff
@@ -391,6 +395,13 @@ def main_event_id=(event_id)
     super(event_id.blank? ? nil : event_id)
   end
 
+  def events_with_rounds
+    competition_events
+      .includes({ rounds: [:competition_event] })
```

Comment:
Nitpick: You don't even need the outer `{}` braces. The `.includes` invocation supports kwargs.